test/test_components.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+import pytest
+
+import pypsa
+
+
+@pytest.fixture
+def network():
+    return pypsa.examples.ac_dc_meshed(from_master=True)
+
+
+@pytest.fixture
+def empty_network_5_buses():
+    # Set up empty network with 5 buses.
+    network = pypsa.Network()
+    n_buses = 5
+    for i in range(n_buses):
+        network.add(
+            'Bus',
+            f'bus_{i}'
+        )
+    return network
+
+
+def test_mremove(network):
+    """
+    GIVEN   the AC DC exemplary pypsa network
+
+    WHEN    two components of Generator are removed with mremove
+
+    THEN    the generator dataframe and the time-dependent generator dataframe
+                should not contain the removed elements.
+    """
+    generators = ['Manchester Wind', 'Frankfurt Wind']
+
+    network.mremove('Generator', generators)
+
+    assert generators not in network.generators.index.tolist()
+    assert generators not in network.generators_t.p_max_pu.columns.tolist()
+
+
+def test_mremove_misspelled_component(network, caplog):
+    """
+    GIVEN   the AC DC exemplary pypsa network
+
+    WHEN    a misspelled component is removed with mremove
+
+    THEN    the function should not change anything in the Line component
+                dataframe and an error should be logged.
+    """
+
+    len_lines = len(network.lines.index)
+
+    network.mremove('Liness', ['0', '1'])
+
+    assert len_lines == len(network.lines.index)
+    assert caplog.records[-1].levelname == 'ERROR'
+
+
+def test_mremove_non_component(network, caplog):
+    """
+    GIVEN   the AC DC exemplary pypsa network
+
+    WHEN    a non-existing component is removed with mremove
+
+    THEN    an error should be logged.
+    """
+
+    network = pypsa.examples.ac_dc_meshed(from_master=True)
+
+    network.mremove('ABC', ['0', '1'])
+
+    assert caplog.records[-1].levelname == 'ERROR'
+
+
+def test_madd_static(empty_network_5_buses):
+    """
+    GIVEN   an empty PyPSA network with 5 buses
+
+    WHEN    multiple components of Load are added to the network with madd and
+                attribute p_set
+
+    THEN    the corresponding load components should be in the index of the
+                static load dataframe. Also the column p_set should contain any
+                value greater than 0.
+    """
+
+    buses = empty_network_5_buses.buses.index.tolist()
+
+    # Add load components at every bus with attribute p_set.
+    load_names = "load_" + empty_network_5_buses.buses.index
+    empty_network_5_buses.madd(
+        "Load",
+        load_names,
+        bus=buses,
+        p_set=np.random.rand(len(buses)),
+    )
+
+    assert load_names.equals(empty_network_5_buses.loads.index)
+    assert any(empty_network_5_buses.loads.p_set > 0)
```

Comment:
```suggestion
        p_set=3,
    )

    assert load_names.equals(empty_network_5_buses.loads.index)
    assert (empty_network_5_buses.loads.p_set == 3).all()
```
Let's strengthen the assertion test a bit.